packages/core/src/combinator/errors.ts

```diff
@@ -10,84 +10,95 @@ import { empty, isCanonicalEmpty } from '../source/empty'
 import { propagateErrorTask } from '../scheduler/PropagateTask'
 import { run } from '../run'
 import { withLocalTime } from './withLocalTime'
+import { Stream, Sink, Scheduler, Disposable, Time } from '@most/types'
 
 /**
  * If stream encounters an error, recover and continue with items from stream
  * returned by f.
- * @param {function(error:*):Stream} f function which returns a new stream
- * @param {Stream} stream
- * @returns {Stream} new stream which will recover from an error by calling f
+ * @param f which returns a new stream
+ * @param stream
+ * @returns new stream which will recover from an error by calling f
  */
-export const recoverWith = (f, stream) =>
+export const recoverWith = <A, E extends Error>(f: (error: E) => Stream<A>, stream: Stream<A>): Stream<A> =>
   isCanonicalEmpty(stream) ? empty()
     : new RecoverWith(f, stream)
 
 /**
  * Create a stream containing only an error
- * @param {*} e error value, preferably an Error or Error subtype
- * @returns {Stream} new stream containing only an error
+ * @param e error value, preferably an Error or Error subtype
+ * @returns new stream containing only an error
  */
-export const throwError = e =>
+export const throwError = (e: Error): Stream<never> =>
```

Comment:
I don’t see how `throwError = <E>(e: E)` would be a breaking change to the types. All it does is relaxing the constraint. Thus, passing an `Error` would still work.

Reply:
If `E` does not extend `Error` then it does not satisfy `propagateErrorTask`/`propagateTask`+`runError` because they require an `Error` to be passed to `Sink#error`.

Reply:
If `Sink#error` is also relaxed then all `Error` constraints can be removed.

Reply:
Thanks for looking into that, @raveclassic. I am curious what @briancavalier has to say to that.

Reply:
That's a bummer, but makes sense.  I don't think we should block this PR for it.  We've talked about parameterizing the error type of Sink, so that's something we can tackle later.